exec/jdbc/src/main/java/org/apache/drill/jdbc/impl/DrillPreparedStatementImpl.java

```diff
@@ -552,16 +701,22 @@ public void clearParameters() throws SQLException {
   public boolean execute() throws SQLException {
     throwIfClosed();
     try {
+      if (timeoutTrigger != null) {
+        timeoutTriggerHandle = timeoutTrigger.startCountdown();
+      }
       return super.execute();
     }
     catch (UnsupportedOperationException e) {
+      if ( isTimedOut() ) {
```

Comment:
i'm not sure it makes sense to check it query timed out if operation is unsupported